pkg/istio-agent/xds_proxy.go

```diff
@@ -420,11 +421,13 @@ func (p *XdsProxy) buildUpstreamClientDialOpts(sa *Agent) ([]grpc.DialOption, er
 	// In these cases, while we fallback to mTLS to istiod using the provisioned certs
 	// it would be ideal to keep using token plus k8s ca certs for control plane communication
 	// as the intention behind provisioned certs on k8s pods is only for data plane comm.
-	if sa.secOpts.ProvCert == "" || !sa.secOpts.FileMountedCerts {
-		dialOptions = append(dialOptions, grpc.WithPerRPCCredentials(oauth.TokenSource{TokenSource: &fileTokenSource{
-			sa.secOpts.JWTPath,
-			time.Second * 300,
-		}}))
+	if sa.proxyConfig.ControlPlaneAuthPolicy != meshconfig.AuthenticationPolicy_NONE {
+		if sa.secOpts.ProvCert == "" || !sa.secOpts.FileMountedCerts {
+			dialOptions = append(dialOptions, grpc.WithPerRPCCredentials(oauth.TokenSource{TokenSource: &fileTokenSource{
```

Comment:
I am not sure why perrpc token has something to do with tls.

From the doc i can see `AuthenticationPolicy_NONE`  indicates the connection is plaintext. It does not say that no token auth.